test/python/compiler/test_transpiler.py

```diff
@@ -719,6 +719,23 @@ def test_block_collection_runs_for_non_cx_bases(self, basis_gates):
 
         self.assertLessEqual(out.count_ops()[twoq_gate], 2)
 
+    @unpack
+    @data(
+        (['u3', 'cx'], {'u3': 1, 'cx': 1}),
+        (['rx', 'rz', 'iswap'], {'rx': 6, 'rz': 11, 'iswap': 2}),
+        (['rx', 'ry', 'rxx'], {'rx': 6, 'ry': 5, 'rxx': 1}),
+    )
+    def test_block_collection_reduces_1q_gate(self, basis_gates, gate_counts):
+        """For synthesis to non-U3 bases, verify we minimize 1q gates."""
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0, 1)
+
+        out = transpile(qc, basis_gates=basis_gates, optimization_level=3)
+
+        self.assertTrue(Operator(out).equiv(qc))
```

Comment:
```suggestion
        self.assertTrue(Operator(out).equiv(Operator(qc)))
```

Reply:
Is this needed, or for clarity? `equiv` starts with a `Operator(other)` if the input supports it.

Reply:
oh I didn't know that function is overloaded.